botocore/credentials.py

```diff
@@ -1031,9 +1064,28 @@ def load(self):
                             self._creds_filename)
                 access_key, secret_key = self._extract_creds_from_mapping(
                     config, self.ACCESS_KEY, self.SECRET_KEY)
+
                 token = self._get_session_token(config)
-                return Credentials(access_key, secret_key, token,
-                                   method=self.METHOD)
+                credentials = Credentials(access_key, secret_key, token, method=self.METHOD)
```

Comment:
would you consider also allowing `source_profile` in this case? This could be useful if an IAM user had some permissions that didn't require an MFA device and others that did e.g.

```
[default]
region = us-west-2
aws_access_key_id = YYY1
aws_secret_access_key = XXX1

[admin-stuff]
region = us-west-2
source_profile = default
mfa_serial = ZZZ1
```

Reply:
@outcoldman looks like this PR is stale pending your response. 

This would be great to have merged!

Reply:
@swordfish444 as I understand none of the maintainers of this repo looked in this PR yet. 

Not sure yet how to proceed. I ended up to use assume_role in my configs, so this PR does not block me anymore.